### PR TITLE
Static nested classes

### PR DESCRIPTION
**Pull request by res0nance**
Minor cleanups to use static nested classes where possible

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] JIRA issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a JIRA issue should exist and be labeled as `lts-candidate`

### CODE REVIEW

**Comment by varyvol** [comment]
@res0nance is the build working for you locally? It failed in CI but the error is quite weird.

**Comment by oleg-nenashev** [comment]
```
[2020-02-19T07:55:53.695Z] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:testCompile (default-testCompile) on project jenkins-test: Compilation failure

[2020-02-19T07:55:53.695Z] [ERROR] An exception has occurred in the compiler (11.0.5). Please file a bug against the Java compiler via the Java bug reporting page (http://bugreport.java.com) after checking the Bug Database (http://bugs.java.com) for duplicates. Include your program and the following diagnostic in your report. Thank you.

[2020-02-19T07:55:53.695Z] [ERROR] java.lang.NullPointerException
```

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Changing public classes to static ones actually breaks the binary compatibility. Also it may lead to funny results when classes are serialized to the disk via XStream. My recommendation would be to avoid changes in public API unless there are issues associated with that. Changing private and [properly] restricted classes is fine

**Comment by res0nance** [comment]
> @res0nance is the build working for you locally? It failed in CI but the error is quite weird.

It does compile on my machine(strangely enough) 🤷‍♂ . It seems to be due to tests compiling but I did not run the test suite. I'll since its an ICE I'll file a bug on the java bug tracker for it.

File: test/src/test/java/jenkins/security/stapler/DoActionFilterTest.java [hunk 1]
```diff
@@ -73,7 +73,7 @@
         
         private TestAccessModifier getPrivate() {return new TestAccessModifier();}
         
-        public class TestAccessModifier {
+        public static class TestAccessModifier {
```
Comment (res0nance) [inline test/src/test/java/jenkins/security/stapler/DoActionFilterTest.java]: Its public here but this is a test class so I just left it

**Review by varyvol — APPROVED** [review]

**Review by MRamonLeon — APPROVED** [review]
If the CI passes

**Comment by MRamonLeon** [comment]
@oleg-nenashev I guess after the changes done you would be fine with the PR, right?

**Review by rsandell — APPROVED** [review]

**Pull request merged**